File: app/src/main/java/com/forkmang/activity/DrawerActivity.java
package com.forkmang.activity;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.drawerlayout.widget.DrawerLayout;

import android.content.Context;
import android.os.Bundle;
import android.view.Menu;
import android.widget.FrameLayout;
import android.widget.Toast;

import com.forkmang.R;
import com.forkmang.fragment.ScanOrderFragment;
import com.google.android.material.navigation.NavigationView;
import com.google.android.material.shape.CornerFamily;
import com.google.android.material.shape.MaterialShapeDrawable;

public class DrawerActivity extends AppCompatActivity {

    Context mContext = DrawerActivity.this;
    public NavigationView navigationView;
    public DrawerLayout drawer;
    public ActionBarDrawerToggle drawerToggle;
    protected FrameLayout frameLayout;


    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drawer);
        frameLayout = findViewById(R.id.content_frame);
        navigationView = findViewById(R.id.nav_view);
        drawer = findViewById(R.id.drawer_layout);

        //View header = navigationView.getHeaderView(0);

        float radius = getResources().getDimension(R.dimen.roundcorner);
        MaterialShapeDrawable navViewBackground = (MaterialShapeDrawable) navigationView.getBackground();
        navViewBackground.setShapeAppearanceModel(
                navViewBackground.getShapeAppearanceModel()
                        .toBuilder()
                        .setTopLeftCorner(CornerFamily.ROUNDED,radius)
                        .setBottomLeftCorner(CornerFamily.ROUNDED,radius)
                        .build());

        setupNavigationDrawer();
    }

    private void setupNavigationDrawer() {
        Menu nav_Menu = navigationView.getMenu();
        navigationView.setNavigationItemSelectedListener(menuItem -> {
            drawer.closeDrawers();
            switch (menuItem.getItemId()) {
                case R.id.menu_scanorder:
                    getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, new ScanOrderFragment())
                            .commit();
                    break;
                case R.id.menu_walkin:
                    /*getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, new WalkinFragment())
                            .commit();*/

                    break;

                case R.id.menu_logout:
                    Toast.makeText(this, "Register Pressed", Toast.LENGTH_SHORT).show();

                    break;

                case R.id.menu_order:
                    //Toast.makeText(this, "Menu Order Pressed", Toast.LENGTH_SHORT).show();
                   /* getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, new OrdersFragment())
                            .commit();*/



                    break;

                case R.id.menu_wallet:
                    Toast.makeText(this, "Menu Wallet Pressed", Toast.LENGTH_SHORT).show();
                    break;

                case R.id.menu_location:
                    Toast.makeText(this, "Menu Location Pressed", Toast.LENGTH_SHORT).show();

                    break;
            }
            return false;
        });
    }


    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        //Sync the toggle state after onRestoreInstanceState has occurred.
        drawerToggle.syncState();
    }



}

File: app/src/main/java/com/forkmang/activity/DrawerActivity.kt
package com.forkmang.activity

import android.os.Bundle
import android.view.MenuItem
import android.widget.FrameLayout
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.drawerlayout.widget.DrawerLayout
import com.forkmang.R
import com.forkmang.databinding.ActivityDrawerBinding
import com.forkmang.fragment.ScanOrderFragment
import com.forkmang.helper.showToastMessage
import com.google.android.material.navigation.NavigationView
import com.google.android.material.shape.CornerFamily
import com.google.android.material.shape.MaterialShapeDrawable

open class DrawerActivity : AppCompatActivity() {

    var navigationView: NavigationView? = null
    var drawer: DrawerLayout? = null
    var drawerToggle: ActionBarDrawerToggle? = null
    protected var frameLayout: FrameLayout? = null

    val binding by lazy { ActivityDrawerBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        frameLayout = binding.contentFrame
        navigationView = binding.navView
        drawer = binding.drawerLayout

  /*      val radius: Float = getResources().getDimension(R.dimen.roundcorner)
           val navViewBackground: MaterialShapeDrawable =
               binding.navView.background as MaterialShapeDrawable
           navViewBackground.shapeAppearanceModel = navViewBackground.shapeAppearanceModel
               .toBuilder()
               .setTopLeftCorner(CornerFamily.ROUNDED, radius)
               .setBottomLeftCorner(CornerFamily.ROUNDED, radius)
               .build()*/
        setupNavigationDrawer()
    }

    private fun setupNavigationDrawer() {
        binding.navView.setNavigationItemSelectedListener { menuItem: MenuItem ->
            binding.drawerLayout.closeDrawers()
            when (menuItem.itemId) {
                R.id.menu_scanorder -> supportFragmentManager.beginTransaction()
                    .replace(R.id.content_frame, ScanOrderFragment())
                    .commit()
                R.id.menu_walkin -> {}
                R.id.menu_logout -> showToastMessage( "Register Pressed")
                R.id.menu_order -> {}
                R.id.menu_wallet -> showToastMessage("Menu Wallet Pressed")
                R.id.menu_location -> showToastMessage("Menu Location Pressed")
            }
            false
        }
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        //Sync the toggle state after onRestoreInstanceState has occurred.
        drawerToggle?.syncState()
    }
}